integrations/gtag/lib/index.js

```diff
@@ -0,0 +1,771 @@
+  .option('awConversionId', '')
+  .option('dcFloodLightId', '')
+  .option('trackAllPages', false)
+  .option('trackNamedPages', true)
+  .option('trackCategorizedPages', true)
+  .option('gaOptions', {
+    classic: false,
+    enhancedEcommerce: false,
+    setAllMappedProps: true
+  })
+  .tag(
+    '<script src="//www.googletagmanager.com/gtag/js?id={{ accountId }}&l=gtagDataLayer">'
+  ));
+
+GTAG.on('construct', function(Integration) {
+  /* eslint-disable */
+  if (Integration.options.gaOptions.classic) {
+      Integration.page = Integration.pageClassic;
+      Integration.orderCompleted = Integration.orderCompletedClassic;
+    } else if (Integration.options.gaOptions.enhancedEcommerce) {
+      Integration.productListViewed = Integration.productListViewedEnhanced;
+      Integration.productClicked = Integration.productClickedEnhanced;
+      Integration.productViewed = Integration.productViewedEnhanced;
+      Integration.productAdded = Integration.productAddedEnhanced;
+      Integration.productRemoved = Integration.productRemovedEnhanced;
+      Integration.promotionViewed = Integration.promotionViewedEnhanced;
+      Integration.promotionClicked = Integration.promotionClickedEnhanced;
+      Integration.checkoutStarted = Integration.checkoutStartedEnhanced;
+      Integration.orderRefunded = Integration.orderRefundedEnhanced;
+      Integration.orderCompleted = Integration.orderCompletedEnhanced;
+      Integration.checkoutStepViewed = Integration.checkoutStepViewedEnhanced;
+      Integration.checkoutStepCompleted =
+        Integration.checkoutStepCompletedEnhanced;
+      Integration.orderUpdated = Integration.orderUpdatedEnhanced;
+
+      // Additional event on top og GA destination
+      Integration.productAddedToWishlist = Integration.productAddedToWishlistEnhanced;
+      Integration.productShared = Integration.productSharedEnhanced;
+
+      // There is no corrosponding event present for following in segment like
+      // loggedIn, timingCompleted, leadGenerated
+
+      // Integration.loggedIn = Integration.loggedInEnhanced;
+      // Integration.signedUp = Integration.signedUpEnhanced;
+      // Integration.exceptionOccured = Integration.exceptionOccuredEnhanced;
+      // Integration.timingCompleted = Integration.timingCompletedEnhanced;
+      // Integration.leadGenerated = Integration.leadGeneratedEnhanced;
+      // Integration.setCheckoutOption = Integration.setCheckoutOptionEnhanced;
+
+      // This mapping is for events which are not supported by segment and will
+      // be tracked by track method.
+      Integration.customEventsMapping = {
+        login: Integration.loggedInEnhanced,
+        sign_up: Integration.signedUpEnhanced,
+        exception: Integration.exceptionOccuredEnhanced,
+        timing_complete: Integration.timingCompletedEnhanced,
+        generate_lead: Integration.leadGeneratedEnhanced,
+        set_checkout_option :Integration.setCheckoutOptionEnhanced
+      };
```

Comment:
I think we have a typo here in the spacing.

Reply:
corrected